src/Microsoft.Data.SqlClient/tests/Common/LocalAppContextSwitchesHelper.cs

```diff
@@ -0,0 +1,355 @@
+#if NETFRAMEWORK
+    public bool DisableTNIRByDefault
+    {
+        get => (bool)_disableTNIRByDefaultProperty.GetValue(null);
+    }
+#endif
+
+    // These properties get or set the like-named underlying switch field value.
+    //
+    // They all fail the test if the value cannot be retrieved or set.
+
+    public Tristate LegacyRowVersionNullBehaviorField
+    {
+        get => GetValue(_legacyRowVersionNullBehaviorField);
+        set => SetValue(_legacyRowVersionNullBehaviorField, value);
+    }
+
+    public Tristate SuppressInsecureTLSWarningField
+    {
+        get => GetValue(_suppressInsecureTLSWarningField);
+        set => SetValue(_suppressInsecureTLSWarningField, value);
+    }
+
+    public Tristate MakeReadAsyncBlockingField
+    {
+        get => GetValue(_makeReadAsyncBlockingField);
+        set => SetValue(_makeReadAsyncBlockingField, value);
+    }
+
+    public Tristate UseMinimumLoginTimeoutField
+    {
+        get => GetValue(_useMinimumLoginTimeoutField);
+        set => SetValue(_useMinimumLoginTimeoutField, value);
+    }
+
+    public Tristate LegacyVarTimeZeroScaleBehaviourField
+    {
+        get => GetValue(_legacyVarTimeZeroScaleBehaviourField);
+        set => SetValue(_legacyVarTimeZeroScaleBehaviourField, value);
+    }
+
+    public Tristate UseCompatProcessSniField
+    {
+        get => GetValue(_useCompatProcessSniField);
+        set => SetValue(_useCompatProcessSniField, value);
+    }
+
+    public Tristate UseCompatAsyncBehaviourField
+    {
+        get => GetValue(_useCompatAsyncBehaviourField);
+        set => SetValue(_useCompatAsyncBehaviourField, value);
+    }
+
+#if NETFRAMEWORK
+    public Tristate DisableTNIRByDefaultField
+    {
+        get => GetValue(_disableTNIRByDefaultField);
+        set => SetValue(_disableTNIRByDefaultField, value);
+    }
+#endif
+
+    #endregion
+
+    #region Private Helpers
+
+    private static Tristate GetValue(FieldInfo field)
+    {
+        var value = field.GetValue(null);
+        if (value is null)
+        {
+            Assert.Fail($"Field {field.Name} has a null value.");
+        }
+
+        return (Tristate)value;
+    }
+
+    private static void SetValue(FieldInfo field, Tristate value)
+    {
+        field.SetValue(null, (byte)value);
+    }
+
+    #endregion
+
+    #region Private Members
+
+    // These fields are used to expose LocalAppContextSwitches's properties.
+    private readonly PropertyInfo _legacyRowVersionNullBehaviorProperty;
+    private readonly PropertyInfo _suppressInsecureTLSWarningProperty;
+    private readonly PropertyInfo _makeReadAsyncBlockingProperty;
+    private readonly PropertyInfo _useMinimumLoginTimeoutProperty;
+    private readonly PropertyInfo _legacyVarTimeZeroScaleBehaviourProperty;
+    private readonly PropertyInfo _useCompatProcessSniProperty;
+    private readonly PropertyInfo _useCompatAsyncBehaviourProperty;
+#if NETFRAMEWORK
+    private readonly PropertyInfo _disableTNIRByDefaultProperty;
```

Comment:
I'm gonna be that guy and say i don't really are what the property is called in the local app context switches, they were written before we took over, and now that we're taking over, we're gonna name things properly (ie, no capitalized acronyms)!

Reply:
Sometimes it do be like that. 😖